engine/table/src/test/java/io/deephaven/engine/table/impl/QueryTableNegativeZeroTest.java

```diff
@@ -0,0 +1,459 @@
+        final Table t2 = newTable(doubleCol("KeyPos", 0.0), doubleCol("Pos", 0.0));
+        for (Table out : List.of(
+                t1.naturalJoin(t2, "KeyNeg=KeyPos"),
+                t2.naturalJoin(t1, "KeyPos=KeyNeg"),
+                t1.exactJoin(t2, "KeyNeg=KeyPos"),
+                t2.exactJoin(t1, "KeyPos=KeyNeg"),
+                t1.join(t2, "KeyNeg=KeyPos"),
+                t2.join(t1, "KeyPos=KeyNeg"),
+                t1.aj(t2, "KeyNeg>=KeyPos"),
+                t2.aj(t1, "KeyPos>=KeyNeg"))) {
+            final long key = oneKey(out);
+            doubleToBitsEquals(out, "KeyNeg", key, -0.0);
+            doubleToBitsEquals(out, "KeyPos", key, 0.0);
+            doubleToBitsEquals(out, "Neg", key, -0.0);
+            doubleToBitsEquals(out, "Pos", key, 0.0);
+        }
+    }
+
+    @Test
+    public void testDhc3768_float_agg_key() {
+        final Table xy = newTable(
+                floatCol("X", -0.0f, 0.0f),
+                intCol("Y", 0, 1),
+                intCol("B", 0, 1));
+        final CreateAggSpecsDhc3768 cas = new CreateAggSpecsDhc3768();
+        AggSpec.visitAll(cas);
+
+        for (AggSpec aggSpec : cas.out) {
+            TestCase.assertEquals(1, xy.aggAllBy(aggSpec, "X").size());
+        }
+
+        {
+            final List<Aggregation> allAggregations = new ArrayList<>();
+            for (AggSpec aggSpec : cas.out) {
+                allAggregations.add(Aggregation.of(aggSpec, String.format("Out_%d=Y", allAggregations.size())));
+            }
+            allAggregations.add(Aggregation.AggCount("Count"));
+            allAggregations.add(Aggregation.AggFirstRowKey("FirstRowKey"));
+            allAggregations.add(Aggregation.AggLastRowKey("LastRowKey"));
+            TestCase.assertEquals(1, xy.aggBy(allAggregations, "X").size());
+        }
+    }
+
+    @Test
+    public void testDhc3768_double_agg_key() {
+        final Table xy = newTable(
+                doubleCol("X", -0.0, 0.0),
+                intCol("Y", 0, 1),
+                intCol("B", 0, 1));
+        final CreateAggSpecsDhc3768 cas = new CreateAggSpecsDhc3768();
+        AggSpec.visitAll(cas);
+
+        for (AggSpec aggSpec : cas.out) {
+            TestCase.assertEquals(1, xy.aggAllBy(aggSpec, "X").size());
+        }
+
+        {
+            final List<Aggregation> allAggregations = new ArrayList<>();
+            for (AggSpec aggSpec : cas.out) {
+                allAggregations.add(Aggregation.of(aggSpec, String.format("Out_%d=Y", allAggregations.size())));
+            }
+            allAggregations.add(Aggregation.AggCount("Count"));
+            allAggregations.add(Aggregation.AggFirstRowKey("FirstRowKey"));
+            allAggregations.add(Aggregation.AggLastRowKey("LastRowKey"));
+            TestCase.assertEquals(1, xy.aggBy(allAggregations, "X").size());
+        }
+    }
+
+    @Test
+    public void testDhc3768_min_max_result() {
```

Comment:
List of aggs to test:
* min (looking to preserve encounter order)
* max (looking to preserve encounter order)
* percentile
* median
* count distinct
* distinct
* unique
* sorted first
* sorted last

Reply:
I think I've got these bases covered now.